chess2.cpp

```diff
@@ -0,0 +1,882 @@
+#include "chess2.h"
+#include<iostream>
+
+int Chess2::perform(str Move) {
+    int original;
+    original = board[Move[2] - 48][Move[3] - 48];
+    board[Move[2] - 48][Move[3] - 48] = board[Move[0] - 48][Move[1] - 48];
+    board[Move[0] - 48][Move[1] - 48] = 0;
+    return original;
+}
+
+str Chess2::push(int row, int col, int desrow, int descol) {
+    using std::to_string;
+    str mystr = to_string(row) + to_string(col) + to_string(desrow) + to_string(descol);
+    return mystr;
+}
+
+str Chess2::computer_move(unsigned short int depth) {
```

Comment:
1.Use a unique naming conventions.
2.Do proper exception handling in cas minMaxroot function call.
3.Log the param in using debug level.